### PR TITLE
Use getpeer name on client sockets.

### PR DESCRIPTION
**Pull request by sas**
This is important so we display the right information during
reverse-connect.

### CODE REVIEW

File: Sources/Host/Common/Socket.cpp [hunk 1]
```diff
@@ -355,10 +355,10 @@ std::string Socket::port() const {
     return std::string();
   }
 
+  auto func = listening() ? ::getsockname : ::getpeername;
```
Comment (a20012251) [inline Sources/Host/Common/Socket.cpp]: I'd create a method that returns the function, just follow the DRY rule ~~~

File: Sources/Host/Common/Socket.cpp [hunk 2]
```diff
@@ -355,10 +355,10 @@ std::string Socket::port() const {
...
   struct sockaddr_storage ss;
   socklen_t sslen = sizeof(ss);
-  if (::getsockname(_handle, reinterpret_cast<struct sockaddr *>(&ss), &sslen) <
-      0) {
+  if (func(_handle, reinterpret_cast<struct sockaddr *>(&ss), &sslen) < 0) {
```
Comment (a20012251) [inline Sources/Host/Common/Socket.cpp]: there's code duplication
I'd move lines 358, 360 and 361 to a new function, same as above, to follow the DRY rule

**Comment by sas** [comment]
Committed the first commits (the ones that passed travis) as 22144148e80d2a572a01da5bca423b337f1f836e..9f3f60b0889fc0c6c0a4fd66b81051d601db9897.

**Comment by sas** [comment]
Committed the last patch as facdb90caf613b18da4bf51b73d05487a27d518e.

**Pull request closed without merging**